cpp/test/sg/fil_test.cu

```diff
@@ -762,6 +819,11 @@ std::vector<FilTestParams> predict_dense_inputs = {
   FIL_TEST_PARAMS(num_rows = 103, num_cols = 100'000, depth = 5, num_trees = 1,
                   algo = BATCH_TREE_REORG, leaf_algo = CATEGORICAL_LEAF,
                   num_classes = 3),
+  FIL_TEST_PARAMS(leaf_algo = VECTOR_LEAF, num_classes = 2),
+  FIL_TEST_PARAMS(leaf_algo = VECTOR_LEAF, num_trees = 9, num_classes = 20),
+  FIL_TEST_PARAMS(num_rows = 103, num_cols = 100'000, depth = 5, num_trees = 1,
+                  algo = BATCH_TREE_REORG, leaf_algo = VECTOR_LEAF,
+                  num_classes = 3),
```

Comment:
can we have both a classification and vector regression example? altering this should work.
Also, could you add a ~4'000 class, 3 tree example?

Reply:
I think regression is exactly the same as classification - AVG of vector leaf, or have I misunderstood?

Reply:
infer_k would get the same parameters except for output_stride, which is not super critical to cover, but would be nice. @canonizer ?